imod_coupler/drivers/dfm_metamod/exchange.py

```diff
@@ -192,7 +192,7 @@ def set_realised_shortage_msw_mf6(
         shortage_left = shortage + demand_msw
         demand_mf6_negative = self.demand["mf-riv2dflow1d_flux_negative"]
         condition = np.logical_and(
```

Comment:
condition is True when fully realized:
In that case sum_to_dflow>sum_from_dflow when both negative (extraction reported by dflowfm should exceed the extraction requested by the components). But always True when both are positive. Please rename variables to_dflow = requested, from_dflow is realized, also add this line as a comment. The condition is wrong when both are positive (false when sum_to_dflow<sum_from_dflow, but should be true) 
So you could also write:  sum_to_dflow>-abs(sum_from_dflow)